src/api/app/models/kiwi/to_xml.rb

```diff
@@ -0,0 +1,89 @@
+# TODO: Please overwrite this comment with something explaining the model target
+module Kiwi
+  class ToXml
```

Comment:
Minor point but can you make it clear that this is for converting a Kiwi::Image to xml? i.e. like:
```ruby
module Kiwi
  class Image
    class ToXml
```
or
```ruby
module Kiwi
    class ImageToXml
```

Reply:
Bad naming (using a verb + a noun), I would prefer something like `Kiwi::Image::Xml` and instead of a class I also prefer to use a module to include in the `Kiwi::Image` class. That way you don't need to have the `to_xml` twice and also no need to instanciate an object.

If you stil prefer a class then use class methods, no need to create an instance of a class without any internal state, just the image reference, also because you aren't reusing this class in any way.

And also you will need to test this module/class alone.

Reply:
For me I prefer to use a class to model complex operations like this instead of a module with methods. OOP > Functional programming. I dont really see a problem with having to instantiate an instance of ToXml..

IMO a class is a much more powerful tool for modelling the behaviour of Image#to_xml than a set of methods in a module.

Reply:
Agreed on the naming! Either module or class I don't have a strong preference.
@mdeniz regarding the tests: atm this class is already implicit tested by the image specs. We can refactor this in a later PR.

Reply:
How many times it is going to be used codewise? just one... no need to instanciate an object if you don't want to use the internal state too. I don't like it as it is.

Reply:
> How many times it is going to be used codewise? just one...

I don't see how that makes any difference?

> no need to instanciate an object if you don't want to use the internal state too.

This is a class which outputs xml based on a kiwi image so the internal state is the kiwi image. Do you really think that putting a bunch of methods into a file is better than having a class to model behaviour? Are we Ruby developers or C developers? :P

Reply:
> regarding the tests: atm this class is already implicit tested by the image specs. 

Yeah I agree. We are testing the behaviour of Kiwi::Image overall, so the tests shouldn't care if this code is in a class or a module or whatever..

Reply:
> I don't see how that makes any difference?

> This is a class which outputs xml based on a kiwi image so the internal state is the kiwi image. Do you really think that putting a bunch of methods into a file is better than having a class to model behaviour? 

I'm not talking about using a module but at least using a class method instead of an instance method, if the internall state is not changing why do we need an instance for it? is just a decorator method... so no need to have an object living for exporting one xml string based on the image.

> Are we Ruby developers or C developers? :P

I'm a developer... not tied to any language 🗡

Reply:
> if the internall state is not changing why do we need an instance for it?

Because you can write code that is much more readable and easier to work with if you consider that the image is the internal state. The fact that it never changes doesn't matter.

> so no need to have an object living for exporting one xml string based on the image.

Yes you're right theres no need to have an object for this, you can do it just fine with a method in a module or a class method, but the code will be much better if you use a object and instance methods for this. If you can save then state to a instance variable then that means you don't have to always pass the image around from one method to another.

Here are more reasons why we should not use a class method for this. Before you say "but you can find a blog post on the internet to support any claim you have about programming!", I tried to find anything supporting class methods for this kind of thing but could not..

http://blog.codeclimate.com/blog/2012/11/14/why-ruby-class-methods-resist-refactoring/
http://old.mlomnicki.com/programming/ruby/2011/07/20/class-vs-instance-methods.html

> I'm a developer... not tied to any language :dagger:

Good answer :smile:

Reply:
Ok, anyway. At least change that name... is horrible 😄